In [1]:
#Bu kısımda Büyük Veri Analitiğinde çıkacak olan 2.soru ve 3.soru için çözümleme ve raporlama yapacagız.Verilen verilere göre z-score 
#max-min normalizasyon yapacagız.3.soruda ise verilen verinin istatiksel raporlamasını oluşturacağız.

In [2]:
#İlk başta kütüphaneyi import edelim.
import pandas as pd

In [4]:
#Verimizi import edeceğiz.
data=pd.read_csv('ex1data2.txt')

In [5]:
#İlk birkaç satırımızı görüntüleyelim
data.head()

2104  3  399900
0  1600  3  329900
1  2400  3  369000
2  1416  2  232000
3  3000  4  539900
4  1985  4  299900

In [6]:
#Sütun isimlendirmeleri ile verimizi düzenleyelim
columns=['M^2','Room','Price']
data.columns=columns

In [8]:
#Z-score hesaplamasını yapalım
z_score=(data-data.mean())/data.std()
print(f'Z-Score Değeri:{z_score}')

Z-Score Değeri:         M^2      Room     Price
0  -0.495977 -0.226166 -0.073110
1   0.499874 -0.226166  0.236953
2  -0.725023 -1.526618 -0.849457
3   1.246762  1.074287  1.592190
4  -0.016724  1.074287 -0.311010
5  -0.578135 -0.226166 -0.192060
6  -0.711330 -0.226166 -1.111155
7  -0.769836 -0.226166 -1.008057
8  -0.627927 -0.226166 -0.766192
9  -0.072740  1.074287 -0.786025
10  0.001948 -0.226166  0.062493
11 -0.134981 -0.226166 -0.072325
12  3.086597  2.374739  2.860989
13 -0.909255 -0.226166 -0.628210
14  0.375393  1.074287  0.878490
15 -0.844525 -1.526618 -0.311010
16 -0.949089 -0.226166 -1.104010
17  0.760040  1.074287  1.275767
18  1.285351  1.074287  2.060852
19 -0.288093 -0.226166 -0.683720
20 -0.137471 -1.526618 -0.667067
21 -0.490998 -0.226166 -0.763020
22 -0.045355  1.074287 -0.628210
23  2.354646 -0.226166  1.861810
24 -1.118384 -0.226166 -0.707510
25 -0.672741 -0.226166  0.994268
26  0.656720 -0.226166  1.029953
27  0.250911 -0.226166  1.077533
28  0.794895 -0.226166 -0.31

In [9]:
#Max-Min Normalizasyonu yapalım
max_min=(data-data.min())/(data.max()-data.min())
print(f'Max-Min Score Hesaplama:{max_min}')

Max-Min Score Hesaplama:         M^2  Room     Price
0   0.206288  0.50  0.301887
1   0.426917  0.50  0.375660
2   0.155543  0.25  0.117170
3   0.592388  0.75  0.698113
4   0.312466  0.75  0.245283
5   0.188086  0.50  0.273585
6   0.158577  0.50  0.054904
7   0.145615  0.50  0.079434
8   0.177055  0.50  0.136981
9   0.300055  0.75  0.132262
10  0.316602  0.50  0.334151
11  0.286266  0.50  0.302074
12  1.000000  1.00  1.000000
13  0.114727  0.50  0.169811
14  0.399338  0.75  0.528302
15  0.129068  0.25  0.245283
16  0.105902  0.50  0.056604
17  0.484556  0.75  0.622826
18  0.600938  0.75  0.809623
19  0.252344  0.50  0.156604
20  0.285714  0.25  0.160566
21  0.207391  0.50  0.137736
22  0.306122  0.75  0.169811
23  0.837838  0.50  0.762264
24  0.068395  0.50  0.150943
25  0.167126  0.50  0.555849
26  0.461666  0.50  0.564340
27  0.371760  0.50  0.575660
28  0.492278  0.50  0.245283
29  0.272201  0.25  0.339623
30  0.040816  0.00  0.000000
31  0.327634  0.75  0.273585
32  0.630171  0.50 

In [10]:
#Şimdi soru 3 kısmına geçelim

In [11]:
#Verinin istatiksel özetini öncelikle alalım
data.describe()


M^2       Room          Price
count    46.000000  46.000000      46.000000
mean   1998.434783   3.173913  339119.456522
std     803.333019   0.768963  126103.418369
min     852.000000   1.000000  169900.000000
25%    1429.500000   3.000000  249900.000000
50%    1870.000000   3.000000  299900.000000
75%    2284.500000   4.000000  368875.000000
max    4478.000000   5.000000  699900.000000

Rapor:Veri Setinin İstatiksel Özeti
Sayın yönetici,
Veri setimiz üzerinde yapılan istatiksel özet çalışmasında yukarıda ki istatiksel özet elde edilmiştir.Bu özet m2 room ve price özellikleri ve dağılımını görmekteyiz.İstatikler karar alma mekanizmları için faydalı bilgiler sunmaktadır.

1.Genel İstatistik Bilgisi
  *Toplam Gözlem Sayısı:46
  *Tüm değişkenler eksiksiz veri bulunmaktadır
  
2.Değişkenlere Göre Özet Bilgi
a)M^2(Metrekare)
  -Veri setindeki evlerin ortalama m2's 1998.43 m^2
  -Standart Sapması:803.33 m^2 dir
  -Değerlerin  geniş bir dağılım gösterdiği gözlemlenmiştir.
  -Minimum bir evin metrekaresi 852 metrekaredir.
  -Maksimum bir evin metrekaresi 4478 metrekaredir.
  -Bu bize değerlerin geniş bir alanda dağılım yaşadığını göstermektedir.
  
  
  
3.Room(Oda sayısı)
*Ortalama:3.17 oda
*Standart Sapma:0.77
-Bu bize genel olarak oda sayısının 3-4 oda arası değiştiğini göstermektedir.
*Bir ev için maksimum 5 minimum 1 oda bulunmaktadır.


4.Price(Fiyat)
*Ortalama:339.119 $
*Standart:126.103 $
-Bu bize fiyatlar arasında ciddi bir değişkenlik olduğunu göstermektedir
*Minimum bir fiyat:169.000
*Maksimum bir fiyat:699.900
-Elimizde ki verilere baktığımızda konut çeşitliğinin farklılıklar gösterdiğini görmekteyiz bu bize farklı segmentlere göre farklı konutlar olduğunu görüyoruz.

  
5)3. Çeyrekler (Percentiles)
Değişkenlerin %25, %50 (medyan), ve %75 dilimleri aşağıdaki gibidir:
M^2: 1429.5, 1870, 2284.5
Room: 3, 3, 4
Price (USD): 249,900, 299,900, 368,875



Sonuç ve Öneriler
1)Metrekare ve fiyat değişkenleri arasında farklı varyasyonlar bulunmaktadır.Bu,farklı boyut ve segmentlerde ki konutların analize alındıgını görmektedir.

2)Oda sayısı homojen dağılımı söz konusudur.Odalar genellikle 3,4 odalı konutlar ağırlık göstermektedir.

3)Bu bize fiyatlandırmalarda fiyat segmentine göre müşteri seçiminin iyi yapılması gerektiğini gösterir.

Analiz sonuçlarının karar alma sürecinde kullanılabileceğini düşünmekte olup daha detaylı analizler için farklı metriklerin de sürece dahil edilmesini önererek geri dönüşünüze sunmaktayım.
Saygılarımla